pkg/cmd/server/bootstrappolicy/policy.go

```diff
@@ -155,7 +159,7 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 					"replicationcontrollers/status", "resourcequotas", "resourcequotas/status", "securitycontextconstraints", "serviceaccounts", "services",
 					"services/status").RuleOrDie(),
 
-				authorizationapi.NewRule(read...).Groups(appsGroup).Resources("statefulsets", "statefulsets/status").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(appsGroup).Resources("statefulsets", "statefulsets/status", "deployments", "deployments/scale", "deployments/status").RuleOrDie(),
```

Comment:
1. You need similar changes in `AdminRoleName`, `EditRoleName`, `ViewRoleName` wrt do adding deployments to `apps`. 
2. We should drop jobs and horizontalpodautoscalers from extensions, everywhere.